File: RailwationBackEnd/Domain/Entities/Country.cs
namespace Domain.Entities;

public class Country: BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string PhotoUrl { get; set; } = "https://content.unops.org/photos/Generic/Impact/_image2880x1400/Ukraine-GettyImages-622892200.jpg";
    public ICollection<CountryConnection> FromCountryConnections { get; set; }
    public ICollection<CountryConnection> ToCountryConnections { get; set; }
    public ICollection<BorderCrossing> BorderCrossingsAsA { get; set; }
    public ICollection<BorderCrossing> BorderCrossingsAsB { get; set; }

}

File: RailwationClient/src/app/dto/Country.ts
import { BorderCrossing } from "./BorderCrossing";
import { CountryConnection } from "./CountryConnection";

export interface Country {
    id: string,
    name: string,
    photoUrl: string,
    fromCountryConnections: CountryConnection[],
    toCountryConnections: CountryConnection[],
    borderCrossingsAsA: BorderCrossing[],
    borderCrossingsAsB: BorderCrossing[],
}